src/components/Web3Provider/index.tsx

```diff
@@ -0,0 +1,136 @@
+
+  const injectedIsActive = hooks.useSelectedIsActive(injected)
+  const previousInjectedIsActive = usePrevious(injectedIsActive)
+
+  const coinbaseWalletIsActive = hooks.useSelectedIsActive(coinbaseWallet)
+  const previousCoinbaseWalletIsActive = usePrevious(coinbaseWalletIsActive)
+
+  const walletConnectIsActive = hooks.useSelectedIsActive(walletConnect)
+  const previousWalletConnectIsActive = usePrevious(walletConnectIsActive)
+
+  const fortmaticIsActive = hooks.useSelectedIsActive(fortmatic)
+  const previousFortmaticIsActive = usePrevious(fortmaticIsActive)
+
+  const [eagerlyConnectingWallets, setEagerlyConnectingWallets] = useState(new Set())
+
+  useEffect(() => {
+    if (error) {
+      console.error(`web3-react error: ${error}`)
+    }
+  }, [error])
+
+  useEffect(() => {
+    if (walletOverride) {
+      getConnectorForWallet(walletOverride).connectEagerly()
+      setEagerlyConnectingWallets(new Set(walletOverride))
+    } else if (!walletOverrideBackfilled) {
+      WALLETS.filter((wallet) => wallet !== Wallet.FORTMATIC)
+        .map(getConnectorForWallet)
+        .forEach((connector) => connector.connectEagerly())
+      setEagerlyConnectingWallets(new Set(WALLETS))
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    const injectedState: ConnectorState = {
+      isActive: injectedIsActive,
+      previousIsActive: previousInjectedIsActive,
+    }
+    const coinbaseWalletState: ConnectorState = {
+      isActive: coinbaseWalletIsActive,
+      previousIsActive: previousCoinbaseWalletIsActive,
+    }
+    const walletConnectState: ConnectorState = {
+      isActive: walletConnectIsActive,
+      previousIsActive: previousWalletConnectIsActive,
+    }
+    const fortmaticState: ConnectorState = {
+      isActive: fortmaticIsActive,
+      previousIsActive: previousFortmaticIsActive,
+    }
+    const isActiveMap = new Map<Wallet, ConnectorState>([
+      [Wallet.INJECTED, injectedState],
+      [Wallet.COINBASE_WALLET, coinbaseWalletState],
+      [Wallet.WALLET_CONNECT, walletConnectState],
+      [Wallet.FORTMATIC, fortmaticState],
+    ])
+
+    isActiveMap.forEach((state: ConnectorState, wallet: Wallet) => {
+      const { isActive, previousIsActive } = state
+      const isEagerlyConnecting = eagerlyConnectingWallets.has(wallet)
+      if (isActive && !previousIsActive) {
+        // When a user manually sets their new connection, set a wallet override.
+        // Also set an override when they were a user prior to this state being introduced.
+        if (!isEagerlyConnecting || !walletOverrideBackfilled) {
+          dispatch(updateWalletOverride({ wallet }))
+        }
+
+        // Reset the eagerly connecting state.
+        if (isEagerlyConnecting) {
+          eagerlyConnectingWallets.delete(wallet)
+          setEagerlyConnectingWallets(new Set([...eagerlyConnectingWallets]))
```

Comment:
Once we have one eagerly connected wallet, do we need to continue trying to connect to the others? Or can we replace this with an empty set?